### ISSUE TITLE
False positive:  E1130 on double unary operator

### ISSUE THREAD

**Issue by bastian-wattro** [pylint-dev/pylint#8597]
### Bug description

Since 2.17.2 Pylint does not accept using a double negation on django Q objects:

(Reproduced in a venv using Python 3.10.10 with nothing but pylint and django installed)

```python
"""Bug reproduction: false positive E1130"""
from django.db.models import Q

def false_positive(need_text=False) -> Q:
    """note the double negation"""
    has_text = ~Q(text="")
    if need_text:
        return has_text
    return ~has_text # E1130: bad operand type for unary ~: Node (invalid-unary-operand-type)

def this_is_ok(need_text=False) -> Q:
    """no double negation here"""
    has_no_text = Q(text="")
    if need_text:
        return ~has_no_text
    return has_no_text
```

### Command used

```shell
$ pylint reproduce.py
```

### Pylint output

```shell
************* Module reproduce
reproduce.py:9:11: E1130: bad operand type for unary ~: Node (invalid-unary-operand-type)
```

### Expected behavior

Same behaviour as in v2.17.1 (no error)

### Pylint version

```shell
2.17.2
```

### OS / Environment

$ uname --kernel-release 
5.10.174-1-MANJARO

**Comment by DanielNoord** [comment]
Have you reported this against https://github.com/PyCQA/pylint-django? Or are you not using it? If it is the latter, please consider doing so.

Going to close this for now as we have explicitly put support for `django` in a separate plugin and repository.
